### PR TITLE
Save user key file on the proxy server.

### PR DESCRIPTION
**Pull request by blueandgold**
The goal of this PR is to save the user keys on the proxy servers.
- We currently do not know which file exactly to save the user keys to.  Jonathan will follow-up, so saving it temporarily, with a TODO.
- Per DD, we will need to notify the admin if the key distribution fails.  Since we have not figure out what's the best way to do this, this is another TODO.
- I would appreciate some suggestion on testing this.  Since we want to avoid writing unit tests that are coupled to the app code, I am having a hard time to see what useful unit test would be worth writing here.  Instead, I think that we almost need to write a monitoring job that will functionally check this is working correctly. 
- Added exception handling in the ssh client if ssh connection can not be established.
- Since I am not sure what file we are writing to, I just decided to write the user keys to a brand new file on each update.  It's pretty easy to change if we need to concat it instead.
- This is a branch off `jobqueue > logger`, and will need to be submitted after those.

[Review on Reviewable](https://reviewable.io/reviews/uproxy/ufo-management-server-flask/37)

### CODE REVIEW

**Comment by blueandgold** [comment]
@jpevarnek PTAL as well, thanks!

File: ufo/key_distributor.py [hunk 1]
```diff
@@ -44,9 +44,34 @@ def _distribute_key(self, proxy_server, key_string):
...
     client = ssh_client.SSHClient()
-    client.connect(proxy_server)
 
-    # TODO do stuff
+    try:
+      client.connect(proxy_server)
+    except (ssh_client.SSHConnectionException, 
+            ssh_client.InvalidKeyTypeException):
+      app.logger.error(
+          'Unable to connect to proxy server to distribute keys: %s',
```
Comment (xwsxethan) [inline ufo/key_distributor.py]: Maybe move the %s string by the proxy server so it doesn't look like we're writing the keys out to the logs. Obviously, you are just writing the proxy server's name, but it reads kind of funny here.

**Comment by xwsxethan** [comment]
This looks fine to me overall. You may want to pull out the file name to a constant so it's easier to find later, but that's about the smallest nit I could pick. I really like the exception handling.

Per your question about testing, I think monitoring makes sense here as well. Whatever time interval we have between syncs, X, if a server's key file has not been touched in say 2X, then the test has failed. You could force some exceptions to be thrown in a unit test and assert that they are handled as we expect, but that's a stretch. I'll leave that up to you. :+1: for me (pending the prior PR of course).

File: ufo/key_distributor.py (hunk 1 above)
Comment (blueandgold) [inline ufo/key_distributor.py]: Done.  Reads much nicer now.

**Comment by blueandgold** [comment]
Thanks for the review!  Both of your comments are addressed.

**Pull request merged**